pkg/block/blocktest/adapter.go

```diff
@@ -463,3 +347,48 @@ func testAdapterWalker(t *testing.T, adapter block.Adapter, storageNamespace str
 		})
 	}
 }
+
+// Test request for a presigned URL for temporary access
+func testGetPreSignedURL(t *testing.T, adapter block.Adapter, storageNamespace string) {
+	ctx := context.Background()
+	obj, _ := objPointers(storageNamespace)
+	expectedExpiry := time.Now().Add(block.DefaultPreSignExpiryDuration)
+
+	preSignedURL, exp, err := adapter.GetPreSignedURL(ctx, obj, block.PreSignModeRead)
+	if adapter.BlockstoreType() == block.BlockstoreTypeGS {
+		require.ErrorContains(t, err, "no credentials found")
+		return
+	} else if adapter.BlockstoreType() == block.BlockstoreTypeLocal {
+		require.ErrorContains(t, err, "operation not supported")
+		return
+	}
+	require.NoError(t, err)
+	require.LessOrEqual(t, exp.Sub(expectedExpiry).Seconds(), 1.0)
```

Comment:
I don't love that this test has an inherent race condition. I realize that one second is a lot relative to the expected time scale, but it has a failure mode that does not depend on the test itself or the unit under test. GitHub runners can sometimes be relatively slow.

I can't think of a way to mitigate this completely. However, moving the declaration of `expectedExpiry` to after `GetPreSignedURL` will likely minimize this risk well enough.